dbcrossbarlib/src/drivers/postgres_shared/table.rs

```diff
@@ -7,6 +7,27 @@ use crate::common::*;
 use crate::schema::Column;
 use crate::separator::Separator;
 
+/// Should we check the PostgreSQL catalog for a schema, or just use the one we
+/// were given?
+///
+/// This is basically a fancy boolean that exists in order to make the related
+/// logic clear at a glance, and easy to verify.
+pub(crate) enum CheckCatalog {
+    /// Check the PostgreSQL catalog for an existing schema.
+    Yes,
+    /// Always use the schema given by the user.
+    No,
+}
+
+impl From<&IfExists> for CheckCatalog {
+    fn from(if_exists: &IfExists) -> CheckCatalog {
+        match if_exists {
+            IfExists::Error | IfExists::Overwrite => CheckCatalog::No,
```

Comment:
hmm but... isn't it useful to pre-check the catalog in _any_ mode, since you might learn things that help you transform cols?

Reply:
@seamusabshere In `--overwrite` mode, we want to ignore any pre-existing schema, because it might have totally different columns. So we don't even look at the existing destination table; we just delete it.